app-typescript/components/TimePickerV2.tsx

```diff
@@ -0,0 +1,166 @@
+import * as React from 'react';
+import nextId from "react-id-generator";
+import { InputWrapper } from './Form';
+import {IInputWrapper} from './Form/InputWrapper';
+import {range} from 'lodash';
+
+interface IProps extends IInputWrapper {
+    value: string;
+    allowSeconds?: boolean;
+    disabledOptions: {
+        hours?: Array<number>;
+        minutes?: Array<number>;
+        seconds?: Array<number>;
+    };
+    'data-test-id'?: string;
+    onChange(valueNext: string): void;
+}
+
+type ITimeUnit = 'hours' | 'minutes' | 'seconds';
+
+export class TimePickerV2 extends React.PureComponent<IProps> {
+    private htmlId = nextId();
+    private hoursStringified = this.stringifyArray(range(25), 'hours');
+    private minutesStringified = this.stringifyArray(range(60), 'minutes');
+    private secondsStringified = this.stringifyArray(range(60), 'seconds');
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.handleChangeForTimeUnit = this.handleChangeForTimeUnit.bind(this);
+        this.stringifyArray = this.stringifyArray.bind(this);
+        this.zeroPad = this.zeroPad.bind(this);
+    }
+
+    stringifyArray(array: Array<number>, timeUnit: ITimeUnit): Array<string> {
+        return array
+            .filter((item) => !(this.props.disabledOptions[timeUnit] ?? []).includes(item))
+            .map((value) => this.zeroPad(value));
+    }
+
+    zeroPad(value: number) {
+        if (value.toString().length === 1 || value === 0) {
+            return `0${value}`;
+        } else if (!value) {
+            return '00';
+        } else {
+            return value.toString();
+        }
+    }
+
+    updateValue(timeUnit: ITimeUnit, stringifyArr: Array<string>): string {
+        const dividedValue = this.props.value.split(':');
+        let value;
+
+        if (timeUnit === 'hours') {
+            value = dividedValue[0];
+        } else if (timeUnit === 'minutes') {
+            value = dividedValue[1];
+        } else {
+            value = dividedValue[2];
+        }
+
+        if (!this.props.disabledOptions[timeUnit]?.includes(Number(value)) && value != null) {
+            return value;
+        } else {
+            return stringifyArr[0];
+        }
+    }
+
+    componentDidMount(): void {
+        if (this.props.allowSeconds) {
+            this.props.onChange(`${this.updateValue('hours', this.hoursStringified)}:${this.updateValue('minutes', this.minutesStringified)}:${this.updateValue('seconds', this.secondsStringified)}`);
```

Comment:
I noticed linter doesn't catch too long lines in some cases. I found it's because of a linter version. @dzonidoo can you install `tslint@5.11.0` on develop (and fix new lint issue)?

Reply:
forgot to mention to do this on develop, not this PR